quickwit/quickwit-control-plane/src/control_plane_model.rs

```diff
@@ -18,37 +18,74 @@
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
 use std::collections::hash_map::Entry;
-use std::time::Instant;
+use std::time::{Duration, Instant};
 
 use anyhow::bail;
 use fnv::{FnvHashMap, FnvHashSet};
 #[cfg(test)]
 use itertools::Itertools;
+use quickwit_common::rate_limiter::{RateLimiter, RateLimiterSettings};
+use quickwit_common::tower::ConstantRate;
 use quickwit_common::Progress;
 use quickwit_config::SourceConfig;
+use quickwit_ingest::{RateMibPerSec, ShardInfo, ShardInfos};
 use quickwit_metastore::{IndexMetadata, ListIndexesMetadataResponseExt};
 use quickwit_proto::control_plane::ControlPlaneResult;
 use quickwit_proto::ingest::{Shard, ShardState};
 use quickwit_proto::metastore::{
     self, EntityKind, ListIndexesMetadataRequest, ListShardsSubrequest, MetastoreError,
     MetastoreService, MetastoreServiceClient, SourceType,
 };
-use quickwit_proto::types::{IndexId, IndexUid, NodeId, NodeIdRef, ShardId, SourceId, SourceUid};
+use quickwit_proto::types::{IndexId, IndexUid, NodeId, ShardId, SourceId, SourceUid};
 use serde::Serialize;
 use tracing::{error, info, warn};
 
+const AUTOSCALING_RATE_LIMITER_SETTINGS: RateLimiterSettings = RateLimiterSettings {
```

Comment:
These values are somewhat arbitrary right now and will require some tuning. It also downscales the number of shards very aggressively; maybe we want one rate limiter for scaling out and another for scaling in?